### PR TITLE
[MB-1594] Package latest Android SDK

### PR DESCRIPTION
**Pull request by rlepinski**
Changes:
 - Reorganized the repo
 - Using JarResolver for Android dependencies
 - Manifest applicationId is now auto resolved during asset loading
 - Menu to regenerate the manifests incase the bundle id changes 
 - Build now packages everything into a unity package to make it easier to install
 - AirshipConfigs are no longer generated for the user. Will work on that in the next PR.

### CODE REVIEW

File: Assets/Plugins/Android/AndroidManifest.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest package="${applicationId}"
+<manifest package="com.example.unity"
```
Comment (marc-scig) [inline Assets/Plugins/Android/AndroidManifest.xml]: I'm sure there's a good reason for this

Comment (rlepinski) [inline Assets/Plugins/Android/AndroidManifest.xml]: Unity cant handle applicationId placeholders. This works fine as the package names are basically ignored in the weird unity manifest merger. I am trying to figure out a way to bypass the application to start takeoff. I believe we can do it with some native code.

File: Assets/UrbanAirship/Editor/UAAndroidAssetPostprocessor.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+				StreamReader sr = new StreamReader(manifest);
+				string body = sr.ReadToEnd();
+				sr.Close();
+
+				body = body.Replace("${applicationId}", PlayerSettings.bundleIdentifier);
+
+				using (var wr = new StreamWriter(manifest, false))
+				{
+					wr.Write(body);
```
Comment (marc-scig) [inline Assets/UrbanAirship/Editor/UAAndroidAssetPostprocessor.cs]: does this need to be closed or is that covered by this fancy using directive?

**Comment by marc-scig** [comment]
LGTM. I'll probably need a walkthrough of some of the deeper gradle magic but I see no cause for alarm.

File: Assets/UrbanAirship/Editor/UAAndroidAssetPostprocessor.cs (hunk 2 above)
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAAndroidAssetPostprocessor.cs]: the fancy using should close it

**Pull request merged**